synapse/neuron.py

```diff
@@ -530,45 +534,42 @@ def _onMesgXmit(self, link, mesg):
             self.taskplex.rx(self, newm)
         except Exception as e:
             logger.exception('xmit taskplex error')
+            self.fini()
+
+    def _send_fail(self, exc, exc_info=None):
+        self.link.tx(('fail', {'exception': (repr(exc), exc_info)}))
```

Comment:
We have a newer, fancier way to marshall exception data.  See getexcfo in synapse.common